compiler/src/iree/compiler/Pipelines/Options.h

```diff
@@ -59,11 +59,11 @@ struct InputDialectOptions {
 // Options controlling high level optimizations.
 struct HighLevelOptimizationOptions {
   // Enables const-expr hoisting into globals.
-  bool constExprHoisting = false;
+  bool constExprHoisting = true;
```

Comment:
One issue with this is reentrancy, spotted by [`tools/test/compile_to_continuation.mlir`](https://github.com/openxla/iree/blob/main/tools/test/compile_to_continuation.mlir): https://gist.github.com/ScottTodd/71a071379d9bc8368c2e87614e79d1d0. Maybe we should restrict that pass to not operate on flow/stream/hal dialect ops? Or early exit if any such ops are observed?

Reply:
Looks like the place for such decisions would be https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Dialect/Util/Analysis/Constant/OpOracle.cpp . This could be a little tricky to balance.

Reply:
We can add Util dialect and op interfaces for controlling hoisting ala DialectInlinerInterface, add that to the ops/dialects we own, and register it as an external interface on the ones we don't (linalg, arith, etc).

Reply:
I tried tweaking memory effects on a few ops to use the existing hoisting logic, but an explicit interface specific to inlining/hoisting would give us more direct control. (I was also going to break that out into its own issue/discussion)